ui/src/contexts/Issuer.tsx

```diff
@@ -0,0 +1,128 @@
+import { message } from "antd";
+import {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { getIssuers, identifierParser } from "src/adapters/api/issuers";
+import { useEnvContext } from "src/contexts/Env";
+import { AppError, Issuer, IssuerIdentifier } from "src/domain";
+import { AsyncTask, isAsyncTaskDataAvailable } from "src/utils/async";
+import {
+  getStorageByKey,
+  isAbortedError,
+  makeRequestAbortable,
+  setStorageByKey,
+} from "src/utils/browser";
+import { IDENTIFIER_LOCAL_STORAGE_KEY } from "src/utils/constants";
+
+type IssuerState = {
+  fetchIssuers: (signal: AbortSignal) => void;
+  handleChange: (identifier: IssuerIdentifier) => void;
+  issuerDisplayName: string;
+  issuerIdentifier: IssuerIdentifier;
+  issuersList: AsyncTask<Issuer[], AppError>;
+};
+
+const defaultIssuerState: IssuerState = {
+  fetchIssuers: () => void {},
+  handleChange: () => void {},
+  issuerDisplayName: "",
+  issuerIdentifier: "",
+  issuersList: { status: "pending" },
+};
+
+const IssuerContext = createContext(defaultIssuerState);
+
+export function IssuerProvider(props: PropsWithChildren) {
+  const env = useEnvContext();
+  const [messageAPI, messageContext] = message.useMessage();
+  const [issuersList, setIssuersList] = useState<AsyncTask<Issuer[], AppError>>({
+    status: "pending",
+  });
+  const [issuerIdentifier, setIssuerIdentifier] = useState<IssuerIdentifier>("");
```

Comment:
The selected identity state should be stored in the URL instead of in the context, otherwise, the URLs are not absolute, meaning they do not always point to the same view and data.